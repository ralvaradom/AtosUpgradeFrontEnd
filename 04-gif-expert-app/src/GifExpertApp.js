import { useState } from "react";
import AddCategory from "./components/AddCategory";
import GifGrid from "./components/GifGrid";

function GifExpertApp({ defaultCategories = [] }) {

	const [ categories, setCategories ] = useState(defaultCategories);

	return (
		<>
			<h2>GifExpertApp</h2>
			<AddCategory setCategories={setCategories} />
			<hr />

			<ol>
				{categories.map((item) => (
					<GifGrid
						key={item}
						category={item}
					/>
				))}
			</ol>
		</>
	);
}

export default GifExpertApp;