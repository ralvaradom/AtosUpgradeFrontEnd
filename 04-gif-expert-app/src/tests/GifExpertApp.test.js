import { shallow } from 'enzyme';
import GifExpertApp from "../GifExpertApp";

describe('Pruebas en <GifExpertApp />', () => {

	test('debe mostararse correctamente', () => {
		const wrapper = shallow(<GifExpertApp />);
		expect(wrapper).toMatchSnapshot();
	});

	test('debe mostarar una lista de categorias', () => {
		const categories = ['One punch', 'Dragon Ball'];
		const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find('GifGrid').length).toBe(categories.length);
	});
});