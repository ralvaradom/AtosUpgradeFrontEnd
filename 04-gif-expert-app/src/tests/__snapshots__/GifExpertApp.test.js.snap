// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> debe mostarar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One punch"
      key="One punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en <GifExpertApp /> debe mostararse correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
